lib/spack/spack/util/environment.py

```diff
@@ -787,11 +786,9 @@ def from_environment_diff(
         modified_variables.sort()
 
         def return_separator_if_any(*args):
-            separators = ":", ";"
-            for separator in separators:
-                for arg in args:
-                    if separator in arg:
-                        return separator
+            for arg in args:
+                if os.pathsep in arg:
+                    return os.pathsep
```

Comment:
From the diff it looks like Unix supports both `;` and `:` separators so `os.pathsep` is not sufficient on that platform. 
I think it should be something like:
```suggestion
            separators = [os.pathsep] if IS_WINDOWS else [":", ";"]
            for separator in separators:
                for arg in args:
                    if separator in arg:
                        return separator
```